field.go

```diff
@@ -214,13 +214,29 @@ func (f *Field) AvailableShards() *roaring.Bitmap {
 	return b
 }
 
-// addRemoteAvailableShards merges the set of available shards into the current known set.
-func (f *Field) addRemoteAvailableShards(b *roaring.Bitmap) {
+// AddRemoteAvailableShards merges the set of available shards into the current known set.
+func (f *Field) AddRemoteAvailableShards(b *roaring.Bitmap) {
 	f.mu.Lock()
 	defer f.mu.Unlock()
 	f.remoteAvailableShards = f.remoteAvailableShards.Union(b)
 }
 
+// RemoveAvailableShard removes a shard from the bitmap cache.
+//
+// NOTE: This can be overridden on the next sync so all nodes should be updated.
+func (f *Field) RemoveAvailableShard(v uint64) error {
```

Comment:
we now have a `saveAvailableShards` method which needs to be called along with this method to persist the `remoteAvailableShards` to disk.

Reply:
@benbjohnson did this make sense? I think this still needs to be implemented.

Reply:
@travisturner It was pulled in through a rebase: https://github.com/pilosa/pilosa/blob/2dfe881e9d7fad88b0832e8861db3d0b97caa672/field.go#L233-L240

Is that what you were thinking?

Reply:
In the case where `RemoveAvailableShard()` is called (in `api.DeleteAvailableShard()` and `server.ReceiveMessage()`), I don't think that `saveAvailableShards()` is called. So the representation of `remoteAvailableShards` on disk still includes the shard that was removed. Even if it went through `AddRemoteAvailableShards()`, that method is doing a `Union`, so it wouldn't remove anything anyway.